WcaOnRails/app/webpacker/lib/wca-live/attempts.js

```diff
@@ -81,11 +227,147 @@ export function formatAttemptResult(attemptResult, eventId) {
+  if (solved < attempted / 2 || solved <= 1) {
+    return { solved: 0, attempted: 0, centiseconds: DNF_VALUE };
+  }
+  // See https://www.worldcubeassociation.org/regulations/#H1b
+  // But allow additional (arbitrary) 30 seconds over the limit for possible +2s.
+  if (centiseconds > 10 * 60 * 100 * Math.min(6, attempted) + 30 * 100) {
+    return { solved: 0, attempted: 0, centiseconds: DNF_VALUE };
+  }
+  return { solved, attempted, centiseconds };
+}
+
+/**
+ * Alters the given FM attempt result, so that it conforms to the WCA regulations.
+ */
+export function autocompleteFmAttemptResult(moves) {
+  // See https://www.worldcubeassociation.org/regulations/#E2d1
+  if (moves > 80) return DNF_VALUE;
+  return moves;
+}
+
+/**
+ * Alters the given time attempt result, so that it conforms to the WCA regulations.
+ */
+export function autocompleteTimeAttemptResult(time) {
+  // See https://www.worldcubeassociation.org/regulations/#9f2
+  return roundOver10Mins(time);
+}
+
+/**
+ * Checks the given attempt results for discrepancies and returns
+ * a warning message if some are found.
+ */
+export function attemptResultsWarning(attemptResults, eventId) {
+  const skippedGapIndex = trimTrailingSkipped(attemptResults).indexOf(
+    SKIPPED_VALUE,
+  );
+  if (skippedGapIndex !== -1) {
+    return `You've omitted attempt ${
+      skippedGapIndex + 1
+    }. Make sure it's intentional.`;
+  }
+  if (eventId === '333mbf') {
+    const lowTimeIndex = attemptResults.findIndex((attempt) => {
+      const { attempted, centiseconds } = decodeMbldAttemptResult(attempt);
+      return attempt > 0 && centiseconds / attempted < 30 * 100;
+    });
+    if (lowTimeIndex !== -1) {
+      return `
+        The result you're trying to submit seems to be impossible:
+        attempt ${lowTimeIndex + 1} is done in
+        less than 30 seconds per cube tried.
+        If you want to enter minutes, don't forget to add two zeros
+        for centiseconds at the end of the score.
+      `;
+    }
+  } else {
+    const completeAttempts = attemptResults.filter(isComplete);
+    if (completeAttempts.length > 0) {
+      const bestSingle = Math.min(...completeAttempts);
+      const worstSingle = Math.max(...completeAttempts);
+      const inconsistent = worstSingle > bestSingle * 4;
+      if (inconsistent) {
+        return `
+          The result you're trying to submit seem to be inconsistent.
+          There's a big difference between the best single
+          (${formatAttemptResult(bestSingle, eventId)}) and the worst single
+          (${formatAttemptResult(worstSingle, eventId)}).
+          Please check that the results are accurate.
+        `;
+      }
+    }
+  }
+  return null;
+}
+
+/**
+ * Alters the given attempt results, so that they conform to the given time limit.
+ */
+export function applyTimeLimit(attemptResults, timeLimit) {
+  if (timeLimit === null) return attemptResults;
+  if (timeLimit.cumulativeRoundWcifIds.length === 0) {
```

Comment:
I don't understand what this block does, nor do I know where `cumulativeRoundWcifIds` comes from 🙈

Reply:
Okay, don't bother too much if it's code we borrowed from WCA Live :D